kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -98,18 +98,45 @@ impl<'tcx> GotocCtx<'tcx> {
         v: ConstValue<'tcx>,
         lit_ty: Ty<'tcx>,
         span: Option<&Span>,
-        data: &Allocation,
+        data: &'tcx Allocation,
         start: usize,
         end: usize,
     ) -> Expr {
         if let ty::Ref(_, ref_ty, _) = lit_ty.kind() {
             match ref_ty.kind() {
                 ty::Str => {
+                    // a string literal
+                    // These seem to always start at 0
+                    assert_eq!(start, 0);
+                    // Create a static variable that holds its value
+                    let mem_var =
+                        self.codegen_allocation_auto_imm_name(data, |tcx| tcx.next_global_name());
+
+                    // Extract identifier for static variable.
+                    // codegen_allocation_auto_imm_name returns the *address* of
+                    // the variable, so need to pattern match to extract it.
+                    let ident = match mem_var.value() {
+                        ExprValue::AddressOf(address) => match address.value() {
```

Comment:
nit: You could merge these matches into one.

Reply:
I didn't realize there was a Box in there. It makes things a bit uglier, so what you have might just be cleaner. But here is how you could merge this:

```rust
let ident = match mem_var.value() {
                        ExprValue::AddressOf(Expr {
                            value: box ExprValue::Symbol { identifier },
                            ..
                        }) => identifier,
                        _ => unreachable!("Expecting a symbol for a string literal allocation"),
                    };
```